### PR TITLE
fix(ui): handle failing voting power

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Handle case when fetching voting power is failing.

Closes: #87 

On error, the voting power indicator should look like:

Screenshot 2024-02-29 at 15 41 14
Screenshot 2024-02-29 at 15 41 04

The error message inside voting power modal looks like:

Screenshot 2024-02-29 at 15 41 00

On retry button click, it should enter loading state:

Screenshot 2024-02-28 at 02 20 24

### How to test

Example of proposal with invalid voting power: https://testnet.snapshotx.xyz/#/s-tn:wan-test.eth/proposal/0x56b857b02d573b0ba747333b57cb3dd11df57cc0d1bcc41c3c990466b477c5e8

1. Go to a proposal with a failing voting power / space's proposals page
2. The voting power indicator should have a special styling
3. Opening the voting power modal should display a message, a retry button, and the list of strategies (without vp)
4. Clicking on the RETRY button should trigger a voting power fetching

### To-Do

On voting power error: 

- [x] Voting power indicator should have an error style
- [x] Show warning message inside modal
- [x] Add retry button inside modal to re-trigger voting power fetching
- [ ] ~~List all available strategies, with a red (!) icon instead of voting power~~
- [x] Improve UI

### CODE REVIEW

**Comment by bonustrack** [comment]
- The red color looks too similar to the Reject button, I think we should keep the original color and use a warning red icon
- We shouldn't show the modal empty when retry is going on we can show a loading instead.
image

**Comment by wa0x6e** [comment]
> * The red color looks too similar to the Reject button, I think we should keep the original color and use a warning red icon
> * We shouldn't show the modal empty when retry is going on we can show a loading instead.
> 
> image

Yes, UI not ready yet, focusing on logic

**Comment by bonustrack** [comment]
This has double border
image

I believe icon should be `exclamation` for error, `information-circle` for info

image

For the modal you could use the flash message style like this:

image

**Comment by wa0x6e** [comment]
Still trying to find out best way to fetch the strategies list

**Comment by wa0x6e** [comment]
Screenshots and code updated as per latest comments

**Review by bonustrack — APPROVED** [review]

File: apps/ui/src/components/IndicatorVotingPower.vue [hunk 1]
```diff
@@ -62,8 +69,10 @@ function handleModalOpen() {
         :network-id="networkId"
         :voting-power-symbol="votingPowerSymbol"
         :voting-powers="props.votingPowers"
+        :voting-power-status="status"
         :final-decimals="decimals"
         @close="modalOpen = false"
+        @get-voting-power="$emit('getVotingPower')"
```
Comment (Sekhmet) [inline apps/ui/src/components/IndicatorVotingPower.vue:75]:
```suggestion
        @get-voting-power="emit('getVotingPower')"
```

File: apps/ui/src/networks/types.ts [hunk 2]
```diff
@@ -77,6 +77,12 @@ export type VotingPower = {
   chainId?: number;
 };
 
+export enum VotingPowerStatus {
+  LOADING,
+  SUCCESS,
+  ERROR
+}
+
```
[the remark this answers is not in the document]
Comment (wa0x6e) [reply apps/ui/src/networks/types.ts]: Moved to union

File: apps/ui/src/components/IndicatorVotingPower.vue [hunk 3]
```diff
@@ -62,8 +68,10 @@ function handleModalOpen() {
...
```
Comment (Sekhmet) [inline apps/ui/src/components/IndicatorVotingPower.vue:74]:
```suggestion
        @get-voting-power="emit('getVotingPower')"
```

Comment (wa0x6e) [reply apps/ui/src/components/IndicatorVotingPower.vue]: Yes, it does not work without the `$`

Comment (Sekhmet) [reply apps/ui/src/components/IndicatorVotingPower.vue:74]: We'd have to do `const emit = defineEmits()`. Actually not sure what should we prefer, I'm not huge fan of `$emit` myself, but we can leave it like this for now.

**Review by Sekhmet — APPROVED** [review]
tACK

**Pull request merged**